pkg/operator/controller/route-status/route_status_test.go

```diff
@@ -0,0 +1,469 @@
+	}{
+		{
+			name: "don't clear anything: all IC admissions are valid",
+			routes: routev1.RouteList{
+				Items: []routev1.Route{
+					*newRouteWithLabelWithAdmittedStatuses("foo-route", "ns", "shard-label", "foo-ic", "bar-ic"),
+					*newRouteWithLabelWithAdmittedStatuses("bar-route", "ns", "shard-label", "baz-ic", "foo-ic", "bar-ic"),
+				},
+			},
+			namespace: corev1.NamespaceList{
+				Items: []corev1.Namespace{
+					*newNamespace("ns", "shard-label"),
+				},
+			},
+			ingressController: newIngressControllerWithSelectors("foo-ic", "shard-label", "", "shard-label", ""),
+			expectedRoutes: routev1.RouteList{
+				Items: []routev1.Route{
+					*newRouteWithLabelWithAdmittedStatuses("foo-route", "ns", "shard-label", "foo-ic", "bar-ic"),
+					*newRouteWithLabelWithAdmittedStatuses("bar-route", "ns", "shard-label", "baz-ic", "foo-ic", "bar-ic"),
+				},
+			},
+		},
+		{
+			name: "clear route that is no longer admitted by ingress controller by routeSelectors",
+			routes: routev1.RouteList{
+				Items: []routev1.Route{
+					*newRouteWithLabelWithAdmittedStatuses("foo-route", "ns", "shard-label", "foo-ic", "bar-ic"),
+					*newRouteWithLabelWithAdmittedStatuses("bar-route", "ns", "", "baz-ic", "foo-ic", "bar-ic"),
+				},
+			},
+			namespace: corev1.NamespaceList{
+				Items: []corev1.Namespace{
+					*newNamespace("ns", ""),
+				},
+			},
+			ingressController: newIngressControllerWithSelectors("foo-ic", "shard-label", "", "", ""),
+			expectedRoutes: routev1.RouteList{
+				Items: []routev1.Route{
+					*newRouteWithLabelWithAdmittedStatuses("foo-route", "ns", "shard-label", "foo-ic", "bar-ic"),
+					*newRouteWithLabelWithAdmittedStatuses("bar-route", "ns", "", "baz-ic", "bar-ic"),
+				},
+			},
+		},
+		{
+			name: "clear route that is no longer admitted by ingress controller by routeSelectors expression",
+			routes: routev1.RouteList{
+				Items: []routev1.Route{
+					*newRouteWithLabelWithAdmittedStatuses("foo-route", "ns", "shard-label", "foo-ic", "bar-ic"),
+					*newRouteWithLabelWithAdmittedStatuses("bar-route", "ns", "", "baz-ic", "foo-ic", "bar-ic"),
+				},
+			},
+			namespace: corev1.NamespaceList{
+				Items: []corev1.Namespace{
+					*newNamespace("ns", ""),
+				},
+			},
+			ingressController: newIngressControllerWithSelectors("foo-ic", "", "shard-label", "", ""),
+			expectedRoutes: routev1.RouteList{
+				Items: []routev1.Route{
+					*newRouteWithLabelWithAdmittedStatuses("foo-route", "ns", "shard-label", "foo-ic", "bar-ic"),
+					*newRouteWithLabelWithAdmittedStatuses("bar-route", "ns", "", "baz-ic", "bar-ic"),
+				},
+			},
+		},
+		{
+			name: "clear route that is no longer admitted by ingress controller by namespaceSelectors",
+			routes: routev1.RouteList{
+				Items: []routev1.Route{
+					*newRouteWithLabelWithAdmittedStatuses("foo-route", "ns", "shard-label", "foo-ic", "bar-ic"),
+					*newRouteWithLabelWithAdmittedStatuses("bar-route", "ns2", "", "baz-ic", "foo-ic", "bar-ic"),
```

Comment:
You want the route to have the right label and just the namespace not to, right?  
```suggestion
					*newRouteWithLabelWithAdmittedStatuses("bar-route", "ns2", "shard-label", "baz-ic", "foo-ic", "bar-ic"),
```